src/pages/index.js

```diff
@@ -1,84 +1,188 @@
+  headers: {
+    authorization: "d081ca18-dd96-4ff1-863a-c195fbcda74a",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInformation = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__pic"
+);
+
+const formValidators = {};
+const enableValidation = (validationSettings) => {
+  DOM.forms.forEach((formElement) => {
+    const validator = new FormValidation(validationSettings, formElement);
+    const formName = formElement.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+let section = 0;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cardData, formData]) => {
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (item) => {
+          const cardElement = createCard(item);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+
+    userInformation.setUserInfo(formData);
+    userInformation.setAvatar(formData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleDeleteCardClick(item) {
+  popupWithConfirmation.open();
+  popupWithConfirmation.setSubmitCall(() => {
+    popupWithConfirmation.deleting(true);
+    api
+      .deleteCard(item.getId())
+      .then(() => {
+        item.handleDeleteCard();
+        popupWithConfirmation.deleting(false);
+      })
+      .catch((err) => {
+        console.log("Error:", err);
+      });
+    popupWithConfirmation.close();
+  });
+}
+
+function handleImageLike(item) {
+  const newLikeStatus = !item.isLiked;
+  if (newLikeStatus) {
+    api
+      .likeCard(item.getId())
+      .then((respond) => {
+        item.setLikeStatus(respond.isLiked);
+      })
+      .catch((err) => {
+        console.log("Error:", err);
+      });
+  } else {
+    api
+      .unlikeCard(item.getId())
+      .then((respond) => {
+        item.setLikeStatus(respond.isLiked);
+      })
+      .catch((err) => {
+        console.error("Error:", err);
+      });
+  }
+}
+
+const popupWithConfirmation = new ConfirmationOnClick({
+  popupSelector: "#modal-delete-confirmation",
+});
+popupWithConfirmation.setEventListeners();
+
 // popup with form
 const newCardPopup = new PopupWithForm(
   "#profile-add-modal",
-  handleCardFormSubmit
+  handleCardAddFormSubmit
 );
 newCardPopup.setEventListeners();
+
 // edit popup form
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const popupEditForm = new PopupWithForm("#profile-edit-modal", (formData) => {
-  userInfo.setUserInfo(formData);
-  popupEditForm.close();
-});
+
+const popupEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  ({ name, about }) => {
+    popupEditForm.saving(true);
+    return api.editProfile({ name, about }).then((editProfile) => {
+      popupEditForm.saving(false);
+      userInformation.setUserInfo(editProfile);
+      popupEditForm.close();
+    });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**